Config restore crash

### Steps to reproduce

Restore a json backup file

### Expected behavior

Should restore all settings & subscriptions

### Actual behavior

Crashes

### LibreTube version

0.11.0

### Android version

Android 13

### Other details

```
java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=300760102, result=-1, data=Intent { dat=content://com.android.externalstorage.documents/... flg=0x1 }} to activity {com.github.libretube/com.github.libretube.ui.activities.SettingsActivity}: kotlinx.serialization.json.internal.JsonDecodingException: Unexpected JSON token at offset 18: Expected start of the array '[', but had ':' instead at path: $.customInstances
JSON input: {"customInstances":null,"localPlaylists":null,"l.....
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5347)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5386)
	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2308)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7898)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: kotlinx.serialization.json.internal.JsonDecodingException: Unexpected JSON token at offset 18: Expected start of the array '[', but had ':' instead at path: $.customInstances
JSON input: {"customInstances":null,"localPlaylists":null,"l.....
	at kotlinx.coroutines.SupervisorKt.JsonDecodingException(Supervisor.kt:1)
	at kotlinx.coroutines.SupervisorKt.JsonDecodingException(Supervisor.kt:2)
	at kotlinx.serialization.json.internal.AbstractJsonLexer.fail(AbstractJsonLexer.kt:64)
	at kotlinx.serialization.json.internal.AbstractJsonLexer.fail$default(AbstractJsonLexer.kt:13)
	at kotlinx.serialization.json.internal.AbstractJsonLexer.fail$kotlinx_serialization_json(AbstractJsonLexer.kt:114)
	at kotlinx.serialization.json.internal.AbstractJsonLexer.unexpectedToken(AbstractJsonLexer.kt:42)
	at kotlinx.serialization.json.internal.AbstractJsonLexer.consumeNextToken(AbstractJsonLexer.kt:9)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.beginStructure(StreamingJsonDecoder.kt:41)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.merge(CollectionSerializers.kt:18)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.deserialize(CollectionSerializers.kt:6)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue$1(StreamingJsonDecoder.kt:67)
	at org.chromium.net.UploadDataSink.decodeSerializableElement(UploadDataSink.java:11)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableElement(StreamingJsonDecoder.kt:45)
	at com.github.libretube.obj.BackupFile$$serializer.deserialize(BackupFile.kt:1)
	at com.github.libretube.obj.BackupFile$$serializer.deserialize(BackupFile.kt:2)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue$1(StreamingJsonDecoder.kt:67)
	at kotlinx.serialization.json.JvmStreamsKt.decodeFromStream(JvmStreams.kt:25)
	at com.github.libretube.ui.preferences.BackupRestoreSettings$$ExternalSyntheticLambda9.onActivityResult(R8$$SyntheticClass:48)
	at androidx.activity.result.ActivityResultRegistry.dispatchResult(ActivityResultRegistry.java:45)
	at androidx.activity.ComponentActivity.onActivityResult(ComponentActivity.java:3)
	at androidx.fragment.app.FragmentActivity.onActivityResult(FragmentActivity.java:6)
	at android.app.Activity.dispatchActivityResult(Activity.java:8613)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5340)
	... 13 more
    
```

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

--- a/app/src/main/java/com/github/libretube/obj/BackupFile.kt
+++ b/app/src/main/java/com/github/libretube/obj/BackupFile.kt
@@ -11,12 +11,12 @@ import kotlinx.serialization.Serializable
 
 @Serializable
 data class BackupFile(
-    var watchHistory: List<WatchHistoryItem> = emptyList(),
-    var watchPositions: List<WatchPosition> = emptyList(),
-    var searchHistory: List<SearchHistoryItem> = emptyList(),
-    var localSubscriptions: List<LocalSubscription> = emptyList(),
-    var customInstances: List<CustomInstance> = emptyList(),
-    var playlistBookmarks: List<PlaylistBookmark> = emptyList(),
-    var localPlaylists: List<LocalPlaylistWithVideos> = emptyList(),
-    var preferences: List<PreferenceItem> = emptyList()
+    var watchHistory: List<WatchHistoryItem>? = emptyList(),
+    var watchPositions: List<WatchPosition>? = emptyList(),
+    var searchHistory: List<SearchHistoryItem>? = emptyList(),
+    var localSubscriptions: List<LocalSubscription>? = emptyList(),
+    var customInstances: List<CustomInstance>? = emptyList(),
+    var playlistBookmarks: List<PlaylistBookmark>? = emptyList(),
+    var localPlaylists: List<LocalPlaylistWithVideos>? = emptyList(),
+    var preferences: List<PreferenceItem>? = emptyList()
 )

--- a/app/src/main/java/com/github/libretube/util/BackupHelper.kt
+++ b/app/src/main/java/com/github/libretube/util/BackupHelper.kt
@@ -50,25 +50,25 @@ class BackupHelper(private val context: Context) {
 
         query {
             Database.watchHistoryDao().insertAll(
-                *backupFile.watchHistory.toTypedArray()
+                *backupFile.watchHistory.orEmpty().toTypedArray()
             )
             Database.searchHistoryDao().insertAll(
-                *backupFile.searchHistory.toTypedArray()
+                *backupFile.searchHistory.orEmpty().toTypedArray()
             )
             Database.watchPositionDao().insertAll(
-                *backupFile.watchPositions.toTypedArray()
+                *backupFile.watchPositions.orEmpty().toTypedArray()
             )
             Database.localSubscriptionDao().insertAll(
-                *backupFile.localSubscriptions.toTypedArray()
+                *backupFile.localSubscriptions.orEmpty().toTypedArray()
             )
             Database.customInstanceDao().insertAll(
-                *backupFile.customInstances.toTypedArray()
+                *backupFile.customInstances.orEmpty().toTypedArray()
             )
             Database.playlistBookmarkDao().insertAll(
-                *backupFile.playlistBookmarks.toTypedArray()
+                *backupFile.playlistBookmarks.orEmpty().toTypedArray()
             )
 
-            backupFile.localPlaylists.forEach {
+            backupFile.localPlaylists.orEmpty().forEach {
                 Database.localPlaylistsDao().createPlaylist(it.playlist)
                 val playlistId = Database.localPlaylistsDao().getAll().last().playlist.id
                 it.videos.forEach {

